src/lib/transaction_logic/mina_transaction_logic.ml

```diff
@@ -546,7 +546,17 @@ module type S = sig
     -> ledger
     -> bool Or_error.t
 
-  module For_tests : sig
+    module For_tests : sig
+      module Stack (Elt : sig type t end) : sig
+        type t = Elt.t list
+        val if_ : bool -> then_:'a -> else_:'a -> 'a
+        val empty : unit -> 'a list
+        val is_empty : 'a list -> bool
```

Comment:
It probably should read
```
val empty: unit -> t
val is_empty: t -> bool
```

Reply:
Good catch, thanks! The way I wrote the signature with polymorphic types was not correct.